nrel/hive/initialization/initialize_simulation.py

```diff
@@ -240,14 +240,31 @@ def osm_init_function(
 
     :raises Exception: from IOErrors parsing the road network
     """
-    if config.input_config.road_network_file is None:
-        raise IOError("Must supply a road network file when using the osm_network")
 
-    road_network = OSMRoadNetwork.from_file(
-        sim_h3_resolution=config.sim.sim_h3_resolution,
-        road_network_file=Path(config.input_config.road_network_file),
-        default_speed_kmph=config.network.default_speed_kmph,
-    )
+    if config.input_config.road_network_file:
+        road_network = OSMRoadNetwork.from_file(
+            sim_h3_resolution=config.sim.sim_h3_resolution,
+            road_network_file=config.input_config.road_network_file,
+            default_speed_kmph=config.network.default_speed_kmph,
+        )
+    elif config.input_config.geofence_file:
+        try:
+            import geopandas
+        except ImportError as e:
+            raise ImportError(
+                "Must have geopandas installed if you want to load from geofence file"
+            ) from e
+
+        dataframe = geopandas.read_file(config.input_config.geofence_file)
+        polygon_union = dataframe["geometry"].unary_union
+
+        road_network = OSMRoadNetwork.from_polygon(
+            sim_h3_resolution=config.sim.sim_h3_resolution,
+            default_speed_kmph=config.network.default_speed_kmph,
+            polygon=polygon_union,
+        )
+    else:
+        raise IOError("Must supply a road network file when using the osm_network")
```

Comment:
this is the error message for omission of either road_network or geofence_file. the error should reflect this, perhaps via `raise IOError("Must supply either a road network or geofence file when using the osm_network")`

Reply:
Fixed